### PR TITLE
docs: Add quickstart guide.

### PR DESCRIPTION
**Pull request by aniemetz**

### CODE REVIEW

**Comment by aniemetz** [comment]
https://cvc5.github.io/docs-ci/docs-pr6782/api/cpp/quickstart.html

File: docs/api/cpp/quickstart.rst [hunk 1]
```diff
@@ -0,0 +1,188 @@
+Quickstart Guide
+================
+
+First, create a cvc5 :cpp:class:`Solver <cvc5::api::Solver>` instance:
```
Comment (yoni206) [inline docs/api/cpp/quickstart.rst:4]: Probably not for this PR, but I wonder what do you think about adding details on how to build this example, that is, starting one step before.

File: docs/api/cpp/quickstart.rst [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+Alternatively, ``"QF_ALL"`` enables all logics without quantifiers.
+To optimize the solver's behavior for a more specific logic,
+use the logic name, e.g. ``"QF_BV"`` or ``"QF_AUFBV"``.
+
+.. literalinclude:: ../../../examples/api/cpp/quickstart.cpp
+     :language: cpp
+     :lines: 41
+
+In the following, we will define constraints of reals and integers.
+For this, we first query the solver for the corresponding sorts.
```
Comment (yoni206) [inline docs/api/cpp/quickstart.rst:29]: Is `query` the right word? sounds a bit like a check-sat query or something.

File: docs/api/cpp/quickstart.rst [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+.. literalinclude:: ../../../examples/api/cpp/quickstart.cpp
+     :language: cpp
+     :lines: 45-46
+
+Now, we create two constants ``x`` and ``y`` of sort ``Real``,
+and two constants ``a`` and ``b`` of sort ``Integer``.
```
Comment (yoni206) [inline docs/api/cpp/quickstart.rst:36]: What do you think about adding here a clarification about the word `constant`?
Maybe something like:
```
Notice that these are *symbolic* constants, but not actual values.
```

File: docs/api/cpp/quickstart.rst [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  value for y: 1/6
+  value for x - y: 0
+
+We can convert these values to C++ types using standard conversion functions.
+
+.. literalinclude:: ../../../examples/api/cpp/quickstart.cpp
+     :language: cpp
+     :lines: 119-121
+
+Another way to independently compute the values of ``x`` and ``y`` would be
```
Comment (yoni206) [inline docs/api/cpp/quickstart.rst:115]:
```suggestion
Another way to independently compute the value of ``x-y`` would be
```

Would you mind changing this also on the cpp?

File: docs/api/cpp/quickstart.rst [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+using the C++ minus operator instead of asking the solver.
+However, for more complex terms, it is easier to let the solver do the
+evaluation.
+
+.. literalinclude:: ../../../examples/api/cpp/quickstart.cpp
+     :language: cpp
+     :lines: 119-121
```
Comment (yoni206) [inline docs/api/cpp/quickstart.rst:122]:
```suggestion
     :lines: 127-135
```

File: docs/api/cpp/quickstart.rst [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+This time the asserted formula is unsatisfiable:
+
+.. literalinclude:: ../../../examples/api/cpp/quickstart.cpp
+     :language: cpp
+     :lines: 156-157
+
+.. code:: text
+
+  expected: unsat
+  result: unsat
```
Comment (yoni206) [inline docs/api/cpp/quickstart.rst:161]:
```suggestion
The output will be:
.. code:: text

  expected: unsat
  result: unsat
```

File: docs/api/cpp/quickstart.rst [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+
+We can query the solver for an unsatisfiable core, that is, a subset
+of the assertions that is already unsatisfiable.
+
+.. literalinclude:: ../../../examples/api/cpp/quickstart.cpp
+     :language: cpp
+     :lines: 161-166
```
Comment (yoni206) [inline docs/api/cpp/quickstart.rst:168]:
```suggestion
     :lines: 161-167
```

**Review by yoni206 — COMMENTED** [review]
Looks very good, thanks for this! Some ideas and small fixes inline.

File: docs/api/cpp/quickstart.rst (hunk 1 above)
Comment (aniemetz) [reply docs/api/cpp/quickstart.rst:4]: Hm. This is the quickstart guide for the API, not sure if we should have build instructions there. I'm thinking this should probably rather be mentioned in https://cvc5.github.io/docs/examples/examples.html. @nafur what do you think?

File: docs/api/cpp/quickstart.rst (hunk 2 above)
Comment (aniemetz) [reply docs/api/cpp/quickstart.rst:29]: It is the right word, yes. You ask the solver for the sort.

File: docs/api/cpp/quickstart.rst (hunk 3 above)
Comment (aniemetz) [reply docs/api/cpp/quickstart.rst:36]: Oh, I thought I had copied that from the example. It seems that I didn't

**Review by yoni206 — APPROVED** [review]
Thanks for this, and for the changes.

**Pull request merged**